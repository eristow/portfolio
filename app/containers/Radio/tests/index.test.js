/**
 *
 * Tests for Radio
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from 'configureStore';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { Radio } from '../index';

describe('<Radio />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <Radio dispatch={dispatch} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Radio />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
