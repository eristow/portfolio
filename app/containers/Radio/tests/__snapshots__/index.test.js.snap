// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> Should render and match the snapshot 1`] = `
<div
  class="Radio__Container-sc-1rnfhb6-0 djXuUF"
>
  <p>
    Loading...
  </p>
</div>
`;
